In [1]:
%matplotlib inline
import pandas as pd

* This file contains a running list of all of the schools in the Philadelphia School District.  Each of these schools is divided into a row for "All Grades" and a row for each of the grades served.

In [2]:
enrollment_demographics = pd.read_csv('../data/enrollment_demographics.csv')

In [3]:
print(enrollment_demographics.columns)

Index(['School Year', 'SRC School ID', 'School Name', 'Learning Network',
       'Grade Level', 'Student Enrollment', 'ELL Count', 'ELL Pct',
       'Not ELL Count', 'Not ELL Pct', 'IEP Count', 'IEP Pct', 'Not IEP Count',
       'Not IEP Pct', 'Female Count', 'Female Pct', 'Male Count', 'Male Pct',
       'American Indian Count', 'American Indian Pct', 'Asian Count',
       'Asian Pct', 'Black/African American Count',
       'Black/African American Pct', 'Hispanic Count', 'Hispanic Pct',
       'Multi-Race Count', 'Multi-Race Pct', 'Pacific Islander Count',
       'Pacific Islander Pct', 'White Count', 'White Pct',
       'CEP Economically Disadvantaged Rate'],
      dtype='object')


In [4]:
enrollment_demographics.shape

(1673, 33)

In [5]:
enrollment_demographics.describe

<bound method NDFrame.describe of      School Year  SRC School ID                                  School Name  \
0      2018-2019            101                     John Bartram High School   
1      2018-2019            101                     John Bartram High School   
2      2018-2019            101                     John Bartram High School   
3      2018-2019            101                     John Bartram High School   
4      2018-2019            101                     John Bartram High School   
5      2018-2019            102                West Philadelphia High School   
6      2018-2019            102                West Philadelphia High School   
7      2018-2019            102                West Philadelphia High School   
8      2018-2019            102                West Philadelphia High School   
9      2018-2019            102                West Philadelphia High School   
10     2018-2019            103                    High School of the Future   
11    

In [6]:
enrollment_demographics[['School Name','Grade Level','Student Enrollment']]

School Name Grade Level  \
0                        John Bartram High School          09   
1                        John Bartram High School          10   
2                        John Bartram High School          11   
3                        John Bartram High School          12   
4                        John Bartram High School  All Grades   
5                   West Philadelphia High School          09   
6                   West Philadelphia High School          10   
7                   West Philadelphia High School          11   
8                   West Philadelphia High School          12   
9                   West Philadelphia High School  All Grades   
10                      High School of the Future          09   
11                      High School of the Future          10   
12                      High School of the Future          11   
13                      High School of the Future          12   
14                      High School of the Future  All Grades   
15    Paul Robeson High School for Human Services          09   
16    Paul Robeson High School for Human Services          10   
17    Paul Robeson High School for Human Services          11   
18    Paul Robeson High School for Human Services          12   
19    Paul Robeson High School for Human Services  All Grades   
20                   William L. Sayre High School          09   
21                   William L. Sayre High School          10   
22                   William L. Sayre High School          11   
23                   William L. Sayre High School          12   
24                   William L. Sayre High School  All Grades   
25                       William T. Tilden School          05   
26                       William T. Tilden School          06   
27                       William T. Tilden School          07   
28                       William T. Tilden School          08   
29                       William T. Tilden School  All Grades   
...                                           ...         ...   
1643                    William H. Loesche School          00   
1644                    William H. Loesche School          01   
1645                    William H. Loesche School          02   
1646                    William H. Loesche School          03   
1647                    William H. Loesche School          04   
1648                    William H. Loesche School          05   
1649                    William H. Loesche School  All Grades   
1650                          The Workshop School          09   
1651                          The Workshop School          10   
1652                          The Workshop School          11   
1653                          The Workshop School          12   
1654                          The Workshop School  All Grades   
1655            John Hancock Demonstration School          00   
1656            John Hancock Demonstration School          01   
1657            John Hancock Demonstration School          02   
1658            John Hancock Demonstration School          03   
1659            John Hancock Demonstration School          04   
1660            John Hancock Demonstration School          05   
1661            John Hancock Demonstration School          06   
1662            John Hancock Demonstration School          07   
1663            John Hancock Demonstration School          08   
1664            John Hancock Demonstration School  All Grades   
1665                 Philadelphia Virtual Academy          06   
1666                 Philadelphia Virtual Academy          07   
1667                 Philadelphia Virtual Academy          08   
1668                 Philadelphia Virtual Academy          09   
1669                 Philadelphia Virtual Academy          10   
1670                 Philadelphia Virtual Academy          11   
1671                 Philadelphia Virtual Academy          12   
1672                 Philadelphia Virtual Academy  All Grades   

      S

* Here I wanted to get an idea of the enrollment of students at each school and within each grade. I think it is important to keep in mind the number of students that attend each school in order to better analyze the influence that school size may have.

* This file is an edited version of the Enrollment/Demographics file that was uploaded above.  I streamlined the original data set to just show the 65 schools that are within a three mile radius of Penn's campus.

In [7]:
penn_phl_schools = pd.read_csv('../data/penn_phl_school_ed.csv')

In [8]:
penn_phl_schools.shape

(66, 33)

In [9]:
penn_phl_schools.describe()

SRC School ID  Student Enrollment   ELL Count    ELL Pct  \
count      66.000000           66.000000   66.000000  66.000000   
mean      255.590909          479.090909   38.015152   7.029394   
std       159.736030          196.900344   66.661370  10.229636   
min       101.000000          169.000000    0.000000   0.000000   
25%       137.250000          358.500000    3.000000   0.635000   
50%       230.500000          454.000000    9.000000   2.025000   
75%       264.750000          546.750000   44.500000  10.020000   
max       878.000000         1213.000000  373.000000  45.930000   

       Not ELL Count  Not ELL Pct   IEP Count    IEP Pct  Not IEP Count  \
count      66.000000    66.000000   66.000000  66.000000      66.000000   
mean      441.075758    92.970606   69.803030  16.456061     409.287879   
std       183.033525    10.229636   37.889402   8.959386     203.378598   
min       169.000000    54.070000   10.000000   0.820000     110.000000   
25%       320.500000    89.980000   48.000000  10.302500     258.750000   
50%       415.500000    97.975000   58.500000  15.010000     388.000000   
75%       495.250000    99.365000   83.750000  20.950000     479.000000   
max      1209.000000   100.000000  205.000000  41.020000    1203.000000   

       Not IEP Pct  ...  Black/African American Count  \
count    66.000000  ...                     66.000000   
mean     83.543939  ...                    304.045455   
std       8.959386  ...                    142.641393   
min      58.980000  ...                     12.000000   
25%      79.050000  ...                    190.250000   
50%      84.990000  ...                    324.000000   
75%      89.697500  ...                    400.000000   
max      99.180000  ...                    636.000000   

       Black/African American Pct  Hispanic Count  Hispanic Pct  \
count                   66.000000       66.000000     66.000000   
mean                    69.095152       43.742424      7.885152   
std                     29.396203       68.734340     10.045728   
min                      4.290000        2.000000      0.450000   
25%                     53.857500        8.000000      2.120000   
50%                     82.770000       18.000000      4.315000   
75%                     93.102500       46.500000      9.172500   
max                     96.360000      386.000000     50.990000   

       Multi-Race Count  Multi-Race Pct  Pacific Islander Count  \
count         66.000000       66.000000               66.000000   
mean          26.090909        5.303485                0.242424   
std           20.188136        2.957150                0.556379   
min            4.000000        1.040000                0.000000   
25%           12.250000        2.887500                0.000000   
50%           19.500000        4.575000                0.000000   
75%           33.750000        7.267500                0.000000   
max          116.000000       12.910000                3.000000   

       Pacific Islander Pct  White Count  White Pct  
count             66.000000    66.000000  66.000000  
mean               0.053182    57.469697   9.889848  
std                0.123003   100.436313  15.067081  
min                0.000000     0.000000   0.000000  
25%                0.000000     3.000000   0.902500  
50%                0.000000     7.500000   1.975000  
75%                0.000000    66.250000  11.360000  
max                0.640000   491.000000  65.100000  

[8 rows x 28 columns]

In [10]:
penn_phl_schools.columns

Index(['School Year', 'SRC School ID', 'School Name', 'Learning Network',
       'Grade Level', 'Student Enrollment', 'ELL Count', 'ELL Pct',
       'Not ELL Count', 'Not ELL Pct', 'IEP Count', 'IEP Pct', 'Not IEP Count',
       'Not IEP Pct', 'Female Count', 'Female Pct', 'Male Count', 'Male Pct',
       'American Indian Count', 'American Indian Pct', 'Asian Count',
       'Asian Pct', 'Black/African American Count',
       'Black/African American Pct', 'Hispanic Count', 'Hispanic Pct',
       'Multi-Race Count', 'Multi-Race Pct', 'Pacific Islander Count',
       'Pacific Islander Pct', 'White Count', 'White Pct',
       'CEP Economically Disadvantaged Rate'],
      dtype='object')

In [11]:
penn_phl_schools[['School Name','Student Enrollment', 'CEP Economically Disadvantaged Rate']]

School Name  Student Enrollment  \
0                            John Bartram High School                 534   
1                       West Philadelphia High School                 448   
2         Paul Robeson High School for Human Services                 307   
3                        William L. Sayre High School                 412   
4                            William T. Tilden School                 360   
5                              Motivation High School                 401   
6                                   John Barry School                 654   
7                            William C. Bryant School                 440   
8                          Joseph W. Catharine School                 460   
9                          Benjamin B. Comegys School                 485   
10                             Sadie Alexander School                 576   
11                             Andrew Hamilton School                 499   
12                         Avery D. Harrington School                 477   
13                                Henry C. Lea School                 525   
14                       William C. Longstreth School                 390   
15                            Morton McMichael School                 358   
16                            S. Weir Mitchell School                 534   
17                            Thomas G. Morton School                 567   
18                                Samuel Powel School                 244   
19                                James Rhoads School                 510   
20                           Martha Washington School                 350   
21                             Add B. Anderson School                 481   
22                                 Alain Locke School                 440   
23                         Rudolph Blankenburg School                 422   
24                    Middle Years Alternative School                 272   
25                     South Philadelphia High School                 617   
26                      Benjamin Franklin High School                 425   
27  High School for Creative and Performing Arts S...                 747   
28           Science Leadership Academy Middle School                 270   
29                          Julia R. Masterman School                1213   
..                                                ...                 ...   
36                         William M. Meredith School                 573   
37                               Robert Morris School                 257   
38                      Girard Academic Music Program                 620   
39                            Edwin M. Stanton School                 305   
40                        Albert M. Greenfield School                 665   
41                           Chester A. Arthur School                 272   
42                             Laura W. Waring School                 289   
43                              Andrew Jackson School                 588   
44                              Abram S. Jenks School                 280   
45                          Eliza B. Kirkbride School                 553   
46                          George W. Nebinger School                 468   
47                                 Academy at Palumbo                1060   
48                                   Southwark School                 863   
49                         Science Leadership Academy                 500   
50                           Constitution High School                 385   
51                             Vare-Washington School                 369   
52             High School of Engineering and Science                 932   
53                     Strawberry Mansion High School                 169   
54                             James G. Blaine School                 442   
55                                William Dick School                 498   
56                               Edwar

* I wanted to isolate the CEP rate to get a better understanding of the socioeconomic standing of the student bodies at each of the schools. By placing this alongside student enrollment, it provides each CEP percentage with a better reference point.

In [12]:
penn_phl_schools['CEP Economically Disadvantaged Rate'].value_counts()

100.00    47
50.88      1
64.69      1
88.15      1
80.00      1
99.18      1
31.00      1
68.15      1
52.65      1
44.44      1
93.23      1
95.74      1
98.12      1
87.65      1
96.05      1
69.54      1
25.02      1
24.57      1
94.75      1
.          1
Name: CEP Economically Disadvantaged Rate, dtype: int64

* This code helped me to conclude that in 47 of the 65 schools, that are within three miles of Penn's campus, all of their students are eligible for CEP.  Moving forward this information is going to be important for me to keep in mind, especially when making comparisons to the rest of the schools in the district.

* Below is a look at the schools that have the lowest CEP rates, with William M. Meredith having the lowest at 24% of students qualifying for CEP.

In [13]:
penn_phl_schools.iloc[['10', '27','29','34','36','38','40','49','52']]

School Year  SRC School ID  \
10   2018-2019            128   
27   2018-2019            202   
29   2018-2019            214   
34   2018-2019            234   
36   2018-2019            238   
38   2018-2019            241   
40   2018-2019            247   
49   2018-2019            265   
52   2018-2019            403   

                                          School Name Learning Network  \
10                             Sadie Alexander School        Network 2   
27  High School for Creative and Performing Arts S...        Network 1   
29                          Julia R. Masterman School        Network 1   
34                    General George A. McCall School        Network 3   
36                         William M. Meredith School        Network 3   
38                      Girard Academic Music Program        Network 1   
40                        Albert M. Greenfield School        Network 3   
49                         Science Leadership Academy       Innovation   
52             High School of Engineering and Science        Network 4   

   Grade Level  Student Enrollment  ELL Count  ELL Pct  Not ELL Count  \
10  All Grades                 576         63    10.94            513   
27  All Grades                 747          9     1.20            738   
29  All Grades                1213          4     0.33           1209   
34  All Grades                 711        154    21.66            557   
36  All Grades                 573          8     1.40            565   
38  All Grades                 620          4     0.65            616   
40  All Grades                 665         24     3.61            641   
49  All Grades                 500         11     2.20            489   
52  All Grades                 932         12     1.29            920   

    Not ELL Pct  ...  Black/African American Pct  Hispanic Count  \
10        89.06  ...                       19.97              26   
27        98.80  ...                       53.55              80   
29        99.67  ...                       15.50              64   
34        78.34  ...                       11.11              28   
36        98.60  ...                        7.68              32   
38        99.35  ...                       24.52              35   
40        96.39  ...                       18.05              39   
49        97.80  ...                       36.60              50   
52        98.71  ...                       68.24              86   

    Hispanic Pct  Multi-Race Count  Multi-Race Pct  Pacific Islander Count  \
10          4.51                66           11.46                       2   
27         10.71                34            4.55                       1   
29          5.28               116            9.56                       0   
34          3.94                60            8.44                       0   
36          5.58                74           12.91                       1   
38          5.65                56            9.03                       0   
40          5.86                72           10.83                       0   
49         10.00                23            4.60                       0   
52          9.23                52            5.58                       0   

    Pacific Islander Pct  White Count  White Pct  \
10                  0.35          248      43.06   
27                  0.13          183      24.50   
29                  0.00          491      40.48   
34                  0.00          183      25.74   
36                  0.17          373      65.10   
38                  0.00          255      41.13   
40                  0.00          376      56.54   
49                  0.00          188      37.60   
52                  0.00           58       6.22   

    CEP Economically Disadvantaged Rate  
10                                44.44  
27                                64.69  
29                                31.00  
34                                69.54  
36     

* Continuing in my exploration, I wanted to separate out all of the schools that have less than 400 total students enrolled. I wanted to take a closer look at these schools, and see if they have any other similarities amongst each other, or if any other differences arose in comparison to the schools that have upwards of 400 total students enrolled.

 * Below is a look at the schools that have under 400 students enrolled.

In [14]:
penn_phl_schools.iloc[['2', '18', '64','62', '57', '58', '59', '53', '20', '28', '37', '14','4','24','35','41','42','44','50','51']]

School Year  SRC School ID                                  School Name  \
2    2018-2019            105  Paul Robeson High School for Human Services   
18   2018-2019            139                          Samuel Powel School   
64   2018-2019            856                          The Workshop School   
62   2018-2019            509                     Parkway West High School   
57   2018-2019            447                     Richard R. Wright School   
58   2018-2019            453                         Edward Gideon School   
59   2018-2019            456                     William D. Kelley School   
53   2018-2019            414               Strawberry Mansion High School   
20   2018-2019            142                     Martha Washington School   
28   2018-2019            205     Science Leadership Academy Middle School   
37   2018-2019            239                         Robert Morris School   
14   2018-2019            135                 William C. Longstreth School   
4    2018-2019            113                     William T. Tilden School   
24   2018-2019            158              Middle Years Alternative School   
35   2018-2019            237                   Delaplaine McDaniel School   
41   2018-2019            248                     Chester A. Arthur School   
42   2018-2019            249                       Laura W. Waring School   
44   2018-2019            252                        Abram S. Jenks School   
50   2018-2019            267                     Constitution High School   
51   2018-2019            272                       Vare-Washington School   

   Learning Network Grade Level  Student Enrollment  ELL Count  ELL Pct  \
2         Network 1  All Grades                 307          9     2.93   
18        Network 2  All Grades                 244          1     0.41   
64       Innovation  All Grades                 256          9     3.52   
62       Network 13  All Grades                 319          7     2.19   
57        Network 4  All Grades                 348          2     0.57   
58       Network 11  All Grades                 257          1     0.39   
59        Network 4  All Grades                 367          0     0.00   
53       Network 11  All Grades                 169          0     0.00   
20        Network 2  All Grades                 350          2     0.57   
28       Innovation  All Grades                 270          3     1.11   
37        Network 3  All Grades                 257          2     0.78   
14       Network 12  All Grades                 390         10     2.56   
4        Network 10  All Grades                 360         62    17.22   
24        Network 2  All Grades                 272         27     9.93   
35       Network 12  All Grades                 377          7     1.86   
41       Network 12  All Grades                 272          6     2.21   
42        Network 3  All Grades                 289         17     5.88   
44       Network 10  All Grades                 280         58    20.71   
50        Network 3  All Grades                 385         17     4.42   
51        Network 3  All Grades                 369        118    31.98   

    Not ELL Count  Not ELL Pct  ...  Black/African American Pct  \
2             298        97.07  ...                       94.46   
18            243        99.59  ...                       77.05   
64            247        96.48  ...                       85.55   
62            312        97.81  ...                       93.42   
57            346        99.43  ...                       90.80   
58            256        99.61  ...                       95.33   
59            367       100.00  ...                       93.19   
53            169       100.00  ...                       91.12   
20            348        99.43  ...                       86.29   
28            267        98.89  ...                       62.96   
37            255        99.22  ...                

* Below is a look at the three schools with the lowest enrollment of students.

In [15]:
penn_phl_schools.iloc[['53','63','18']]

School Year  SRC School ID                      School Name  \
53   2018-2019            414   Strawberry Mansion High School   
63   2018-2019            518  Vaux High School: A Big Picture   
18   2018-2019            139              Samuel Powel School   

   Learning Network Grade Level  Student Enrollment  ELL Count  ELL Pct  \
53       Network 11  All Grades                 169          0     0.00   
63       Innovation  All Grades                 211          3     1.42   
18        Network 2  All Grades                 244          1     0.41   

    Not ELL Count  Not ELL Pct  ...  Black/African American Pct  \
53            169       100.00  ...                       91.12   
63            208        98.58  ...                       93.36   
18            243        99.59  ...                       77.05   

    Hispanic Count  Hispanic Pct  Multi-Race Count  Multi-Race Pct  \
53               6          3.55                 4            2.37   
63               6          2.84                 5            2.37   
18               5          2.05                18            7.38   

    Pacific Islander Count  Pacific Islander Pct  White Count  White Pct  \
53                       0                   0.0            3       1.78   
63                       0                   0.0            3       1.42   
18                       0                   0.0           21       8.61   

    CEP Economically Disadvantaged Rate  
53                               100.00  
63                               100.00  
18                                95.74  

[3 rows x 33 columns]

* Below are the three schools with the most students enrolled

In [16]:
penn_phl_schools.iloc[['29','47','52']]

School Year  SRC School ID                             School Name  \
29   2018-2019            214               Julia R. Masterman School   
47   2018-2019            262                      Academy at Palumbo   
52   2018-2019            403  High School of Engineering and Science   

   Learning Network Grade Level  Student Enrollment  ELL Count  ELL Pct  \
29        Network 1  All Grades                1213          4     0.33   
47        Network 1  All Grades                1060         32     3.02   
52        Network 4  All Grades                 932         12     1.29   

    Not ELL Count  Not ELL Pct  ...  Black/African American Pct  \
29           1209        99.67  ...                       15.50   
47           1028        96.98  ...                       43.58   
52            920        98.71  ...                       68.24   

    Hispanic Count  Hispanic Pct  Multi-Race Count  Multi-Race Pct  \
29              64          5.28               116            9.56   
47             112         10.57                53            5.00   
52              86          9.23                52            5.58   

    Pacific Islander Count  Pacific Islander Pct  White Count  White Pct  \
29                       0                   0.0          491      40.48   
47                       0                   0.0          127      11.98   
52                       0                   0.0           58       6.22   

    CEP Economically Disadvantaged Rate  
29                                31.00  
47                                88.15  
52                                68.15  

[3 rows x 33 columns]

* This file provides a "score card" for each of the schools in the Philadelphia District.  The score card is measured across 69 metrics.  I narrowed my search in on this file to only focus on the metrics that are relevant to my research questions.  But this file helps to continue to provide more insight into each of the schools.

In [17]:
metric_scores = pd.read_csv('../data/phl_school_metric_scores_revised.csv')

In [18]:
metric_scores.shape

(321, 337)

In [19]:
metric_scores.columns

Index(['School', 'SRC School ID', 'ULCS Code', 'Report', 'Rpt Type Long',
       'Street Address', 'City', 'State', 'Zip Code', 'Phone Number',
       ...
       'FAFSA Tier', 'Student Survey College & Career Score',
       'Student Survey College & Career Pts Earn',
       'Student Survey College & Career Pts Poss',
       'Student Survey College & Career Pct Earn',
       'Student Survey College & Career Tier',
       'Teach Effect Distinguished Score', 'Teach Effect Instruction Score',
       'Teacher Attendance Score', 'Student Survey Teaching Score'],
      dtype='object', length=337)

In [20]:
metric_scores.describe()

ULCS Code      Zip Code   Enrollment  Notes  Clim Score  \
count   321.000000    321.000000   321.000000    0.0  321.000000   
mean   4502.869159  19131.847352   593.598131    NaN   54.000000   
std    2082.983778     12.841038   334.667406    NaN   25.404724   
min    1010.000000  19102.000000    67.000000    NaN    3.000000   
25%    3321.000000  19123.000000   398.000000    NaN   35.000000   
50%    3424.000000  19133.000000   528.000000    NaN   56.000000   
75%    6230.000000  19143.000000   707.000000    NaN   73.000000   
max    8770.000000  19154.000000  3363.000000    NaN   98.000000   

       Clim Pts Earn  Clim Pts Poss  Attendance (95%+) Score  \
count     321.000000     321.000000               321.000000   
mean       14.739003      27.408100                47.595016   
std         7.328808       4.382329                16.960004   
min         0.700000      20.000000                 7.000000   
25%         9.280000      20.000000                35.000000   
50%        14.700000      30.000000                47.000000   
75%        19.950000      30.000000                59.000000   
max        29.510000      30.000000                95.000000   

       Attendance (95%+) Denom  Attendance (95%+) Pts Earn  ...  OSS Pts Poss  \
count               321.000000                  321.000000  ...    321.000000   
mean                646.308411                    5.081090  ...      3.612150   
std                 361.301521                    3.750015  ...      0.657796   
min                  72.000000                    0.000000  ...      2.500000   
25%                 437.000000                    1.790000  ...      2.500000   
50%                 586.000000                    4.590000  ...      4.000000   
75%                 764.000000                    7.690000  ...      4.000000   
max                3608.000000                   12.000000  ...      4.000000   

       OSS Pct Earn  Parent Survey Climate Score  \
count    321.000000                   321.000000   
mean      62.323988                    49.809969   
std       29.870989                    41.122432   
min        0.000000                     0.000000   
25%       41.000000                     0.000000   
50%       68.000000                    76.000000   
75%       88.000000                    85.000000   
max      100.000000                    99.000000   

       Parent Survey Climate Pts Earn  Parent Survey Climate Pts Poss  \
count                      321.000000                           321.0   
mean                         0.498100                             1.0   
std                          0.411224                             0.0   
min                          0.000000                             1.0   
25%                          0.000000                             1.0   
50%                          0.760000                             1.0   
75%                          0.850000                             1.0   
max                          0.990000                             1.0   

       Parent Survey Climate Pct Earn  Parent Survey Participation Score  \
count                      321.000000                         321.000000   
mean                        49.809969                          17.295950   
std                         41.122432                          16.444498   
min                          0.000000                           0.000000   
25%                          0.000000                           5.000000   
50%                         76.000000                          12.000000   
75%                         85.000000                          25.000000   
max                         99.000000                          83.000000   

       Parent Survey Participation Pts Earn  \
count                            321.000000   
mean                               0.231589   
std                                0.314349   
min                                0.000000   
25%                                0.00000

In [21]:
metric_scores[['School', 'Grades Served', 'Retention Score', 'Grad Score']]

School Grades Served  \
0                             John Bartram High School        '9-12'   
1                        West Philadelphia High School        '9-12'   
2                            High School of the Future        '9-12'   
3          Paul Robeson High School for Human Services        '9-12'   
4                         William L. Sayre High School        '9-12'   
5                             William T. Tilden School         '5-8'   
6                               Motivation High School        '9-12'   
7                                    John Barry School         'K-8'   
8                             William C. Bryant School         'K-8'   
9                           Joseph W. Catharine School         'K-5'   
10                          Benjamin B. Comegys School         'K-8'   
11                              Sadie Alexander School         'K-8'   
12                              Andrew Hamilton School         'K-8'   
13                          Avery D. Harrington School         'K-8'   
14                                 Henry C. Lea School         'K-8'   
15                        William C. Longstreth School         'K-8'   
16                             Morton McMichael School         'K-8'   
17                             S. Weir Mitchell School         'K-8'   
18                             Thomas G. Morton School         'K-5'   
19                                 Samuel Powel School         'K-4'   
20                            John M. Patterson School         'K-4'   
21                                 James Rhoads School         'K-8'   
22                            Martha Washington School         'K-8'   
23                                      Penrose School         'K-8'   
24                              Add B. Anderson School         'K-8'   
25                                  Alain Locke School         'K-8'   
26                          Rudolph Blankenburg School         'K-8'   
27                     Middle Years Alternative School         '5-8'   
28                      South Philadelphia High School        '9-12'   
29                       Benjamin Franklin High School        '9-12'   
..                                                 ...           ...   
291             Sankofa Freedom Academy Charter School         'K-8'   
292           Franklin Towne Charter Elementary School         'K-8'   
293                      Tacony Academy Charter School        '9-12'   
294                      Tacony Academy Charter School         'K-8'   
295                   Aspira Charter School at Stetson         '5-8'   
296                  Mastery Charter School at Harrity         'K-8'   
297                     Mastery Charter School at Mann         'K-6'   
298                  Mastery Charter School at Smedley         'K-6'   
299                Universal Charter School at Bluford         'K-6'   
300                 Universal Charter School at Daroff         'K-8'   
301                 Mastery Charter School at Douglass         'K-8'   
302                     Aspira Charter School at Olney        '9-12'   
303                   Mastery Charter School at Clymer         'K-6'   
304                    Mastery Charter School at Gratz        '9-12'   
305                    Mastery Charter School at Gratz         '7-8'   
306              Universal Charter School at Audenried        '9-12'   
307                   Universal Charter School at Vare         '5-8'   
308                  Lindley Academy Charter at Birney         'K-8'   
309                Mastery Charter School at Cleveland         'K-8'   
310              Universal Charter School at Creighton         'K-8'   
311  The Philadelphia Charter School for Arts and S...         'K-8'   
312  Memphis Street Academy Charter School at J.P. ...         '5-8'   
313                 Universal Charter School at Alcorn         'K-8'   
314                Mastery Charter School at Pastorius         'K-8'   
315             

In [22]:
metric_scores[['School', 'Grades Served', 'Attendance (95%+) Score', 'Attendance (<80%) Score']]

School Grades Served  \
0                             John Bartram High School        '9-12'   
1                        West Philadelphia High School        '9-12'   
2                            High School of the Future        '9-12'   
3          Paul Robeson High School for Human Services        '9-12'   
4                         William L. Sayre High School        '9-12'   
5                             William T. Tilden School         '5-8'   
6                               Motivation High School        '9-12'   
7                                    John Barry School         'K-8'   
8                             William C. Bryant School         'K-8'   
9                           Joseph W. Catharine School         'K-5'   
10                          Benjamin B. Comegys School         'K-8'   
11                              Sadie Alexander School         'K-8'   
12                              Andrew Hamilton School         'K-8'   
13                          Avery D. Harrington School         'K-8'   
14                                 Henry C. Lea School         'K-8'   
15                        William C. Longstreth School         'K-8'   
16                             Morton McMichael School         'K-8'   
17                             S. Weir Mitchell School         'K-8'   
18                             Thomas G. Morton School         'K-5'   
19                                 Samuel Powel School         'K-4'   
20                            John M. Patterson School         'K-4'   
21                                 James Rhoads School         'K-8'   
22                            Martha Washington School         'K-8'   
23                                      Penrose School         'K-8'   
24                              Add B. Anderson School         'K-8'   
25                                  Alain Locke School         'K-8'   
26                          Rudolph Blankenburg School         'K-8'   
27                     Middle Years Alternative School         '5-8'   
28                      South Philadelphia High School        '9-12'   
29                       Benjamin Franklin High School        '9-12'   
..                                                 ...           ...   
291             Sankofa Freedom Academy Charter School         'K-8'   
292           Franklin Towne Charter Elementary School         'K-8'   
293                      Tacony Academy Charter School        '9-12'   
294                      Tacony Academy Charter School         'K-8'   
295                   Aspira Charter School at Stetson         '5-8'   
296                  Mastery Charter School at Harrity         'K-8'   
297                     Mastery Charter School at Mann         'K-6'   
298                  Mastery Charter School at Smedley         'K-6'   
299                Universal Charter School at Bluford         'K-6'   
300                 Universal Charter School at Daroff         'K-8'   
301                 Mastery Charter School at Douglass         'K-8'   
302                     Aspira Charter School at Olney        '9-12'   
303                   Mastery Charter School at Clymer         'K-6'   
304                    Mastery Charter School at Gratz        '9-12'   
305                    Mastery Charter School at Gratz         '7-8'   
306              Universal Charter School at Audenried        '9-12'   
307                   Universal Charter School at Vare         '5-8'   
308                  Lindley Academy Charter at Birney         'K-8'   
309                Mastery Charter School at Cleveland         'K-8'   
310              Universal Charter School at Creighton         'K-8'   
311  The Philadelphia Charter School for Arts and S...         'K-8'   
312  Memphis Street Academy Charter School at J.P. ...         '5-8'   
313                 Universal Charter School at Alcorn         'K-8'   
314                Mastery Charter School at Pastorius         'K-8'   
315             

* Some of the metrics I found relevant and singled out; School Name and the Grades Served, Retention Rate, Graduation Rate, percentage of students attending more than 95% of instructional days, and percentage of students attending less than 80% of instructional days.

* However, I found there too be too many rows with "Not Applicable" or "Data Not Available" that I decided to go back and remove from the original spreadsheet.  I then replaced it with the clean one below.
But, note that, only the high schools' graduation rates were presented.

In [25]:
pd.read_csv('../data/phl_school_metric_scores_revised.csv')

School SRC School ID  \
0                             John Bartram High School           101   
1                        West Philadelphia High School           102   
2                            High School of the Future           103   
3          Paul Robeson High School for Human Services           105   
4                         William L. Sayre High School           110   
5                             William T. Tilden School           113   
6                               Motivation High School           119   
7                                    John Barry School           120   
8                             William C. Bryant School           123   
9                           Joseph W. Catharine School           125   
10                          Benjamin B. Comegys School           126   
11                              Sadie Alexander School           128   
12                              Andrew Hamilton School           129   
13                          Avery D. Harrington School           130   
14                                 Henry C. Lea School           134   
15                        William C. Longstreth School           135   
16                             Morton McMichael School           136   
17                             S. Weir Mitchell School           137   
18                             Thomas G. Morton School           138   
19                                 Samuel Powel School           139   
20                            John M. Patterson School           140   
21                                 James Rhoads School           141   
22                            Martha Washington School           142   
23                                      Penrose School           144   
24                              Add B. Anderson School           146   
25                                  Alain Locke School           147   
26                          Rudolph Blankenburg School           149   
27                     Middle Years Alternative School           158   
28                      South Philadelphia High School           200   
29                       Benjamin Franklin High School           201   
..                                                 ...           ...   
291             Sankofa Freedom Academy Charter School           W99   
292           Franklin Towne Charter Elementary School           X03   
293                      Tacony Academy Charter School           X04   
294                      Tacony Academy Charter School           X04   
295                   Aspira Charter School at Stetson           X06   
296                  Mastery Charter School at Harrity           X07   
297                     Mastery Charter School at Mann           X08   
298                  Mastery Charter School at Smedley           X09   
299                Universal Charter School at Bluford           X10   
300                 Universal Charter School at Daroff           X11   
301                 Mastery Charter School at Douglass           X12   
302                     Aspira Charter School at Olney           X14   
303                   Mastery Charter School at Clymer           X15   
304                    Mastery Charter School at Gratz           X16   
305                    Mastery Charter School at Gratz           X16   
306              Universal Charter School at Audenried           X17   
307                   Universal Charter School at Vare           X18   
308                  Lindley Academy Charter at Birney           X19   
309                Mastery Charter School at Cleveland           X20   
310              Universal Charter School at Creighton           X21   
311  The Philadelphia Charter School for Arts and S...           X22   
312  Memphis Street Academy Charter School at J.P. ...           X23   
313                 Universal Charter School at Alcorn           X24   
314                Mastery Charter School at Pastorius           X26   
315             

In [26]:
metric_scores_rvsd = pd.read_csv('../data/phl_school_metric_scores_revised.csv')

In [27]:
metric_scores_rvsd[['School','Teacher Attendance Score']]

School  \
0                             John Bartram High School   
1                        West Philadelphia High School   
2                            High School of the Future   
3          Paul Robeson High School for Human Services   
4                         William L. Sayre High School   
5                             William T. Tilden School   
6                               Motivation High School   
7                                    John Barry School   
8                             William C. Bryant School   
9                           Joseph W. Catharine School   
10                          Benjamin B. Comegys School   
11                              Sadie Alexander School   
12                              Andrew Hamilton School   
13                          Avery D. Harrington School   
14                                 Henry C. Lea School   
15                        William C. Longstreth School   
16                             Morton McMichael School   
17                             S. Weir Mitchell School   
18                             Thomas G. Morton School   
19                                 Samuel Powel School   
20                            John M. Patterson School   
21                                 James Rhoads School   
22                            Martha Washington School   
23                                      Penrose School   
24                              Add B. Anderson School   
25                                  Alain Locke School   
26                          Rudolph Blankenburg School   
27                     Middle Years Alternative School   
28                      South Philadelphia High School   
29                       Benjamin Franklin High School   
..                                                 ...   
291             Sankofa Freedom Academy Charter School   
292           Franklin Towne Charter Elementary School   
293                      Tacony Academy Charter School   
294                      Tacony Academy Charter School   
295                   Aspira Charter School at Stetson   
296                  Mastery Charter School at Harrity   
297                     Mastery Charter School at Mann   
298                  Mastery Charter School at Smedley   
299                Universal Charter School at Bluford   
300                 Universal Charter School at Daroff   
301                 Mastery Charter School at Douglass   
302                     Aspira Charter School at Olney   
303                   Mastery Charter School at Clymer   
304                    Mastery Charter School at Gratz   
305                    Mastery Charter School at Gratz   
306              Universal Charter School at Audenried   
307                   Universal Charter School at Vare   
308                  Lindley Academy Charter at Birney   
309                Mastery Charter School at Cleveland   
310              Universal Charter School at Creighton   
311  The Philadelphia Charter School for Arts and S...   
312  Memphis Street Academy Charter School at J.P. ...   
313                 Universal Charter School at Alcorn   
314                Mastery Charter School at Pastorius   
315                   Mastery Charter School at Wister   
316                  Global Leadership Academy at Huey   
317                   MaST Community Charter School II   
318                         TECH Freire Charter School   
319                   Independence Charter School West   
320              KIPP West Philadelphia Charter School   

    Teacher Attendance Score  
0                         52  
1                         49  
2                         56  
3                         68  
4                         71  
5                         50  
6                         54  
7                         49  
8                         41  
9                         53  
10                        58  
11                        79  
12                        46  
13               

* Branching off into a new segment, I wanted to separate our the "Teacher Attendance" score.  While not all of the schools had this information available, for those that did I found this information to be really relevant, especially when trying to get a holistic picture of the general culture and environment at each school.

* After revising the original file, I decided that I wanted to make my coding process a little simpler and separate out the schools that are within my selected radius ( no further than three miles outside of Penn's campus; 66 schools) from the remainder of schools in the district.

In [28]:
pd.read_csv('../data/sms_penn_schools.csv')

School  SRC School ID  ULCS Code  \
0                         John Bartram High School            101       1010   
1                    West Philadelphia High School            102       1020   
2      Paul Robeson High School for Human Services            105       1050   
3                     William L. Sayre High School            110       1100   
4                         William T. Tilden School            113       1130   
5                           Motivation High School            119       1190   
6                                John Barry School            120       1200   
7                         William C. Bryant School            123       1230   
8                       Joseph W. Catharine School            125       1250   
9                       Benjamin B. Comegys School            126       1260   
10                          Sadie Alexander School            128       1280   
11                          Andrew Hamilton School            129       1290   
12                      Avery D. Harrington School            130       1300   
13                             Henry C. Lea School            134       1340   
14                    William C. Longstreth School            135       1350   
15                         Morton McMichael School            136       1360   
16                         S. Weir Mitchell School            137       1370   
17                         Thomas G. Morton School            138       1380   
18                             Samuel Powel School            139       1390   
19                             James Rhoads School            141       1410   
20                        Martha Washington School            142       1420   
21                          Add B. Anderson School            146       1460   
22                              Alain Locke School            147       1470   
23                      Rudolph Blankenburg School            149       1490   
24                 Middle Years Alternative School            158       1580   
25                  South Philadelphia High School            200       2000   
26                   Benjamin Franklin High School            201       2010   
27    High School for Creative and Performing Arts            202       2020   
28        Science Leadership Academy Middle School            205       2050   
29                  Julia R. Masterman School H.S.            214       2140   
..                                             ...            ...        ...   
35                 General George A. McCall School            234       2340   
36                      Delaplaine McDaniel School            237       2370   
37                      William M. Meredith School            238       2380   
38                            Robert Morris School            239       2390   
39              Girard Academic Music Program H.S.            241       2410   
40              Girard Academic Music Program M.S.            241       2410   
41                         Edwin M. Stanton School            245       2450   
42                     Albert M. Greenfield School            247       2470   
43                        Chester A. Arthur School            248       2480   
44                          Laura W. Waring School            249       2490   
45                           Andrew Jackson School            251       2510   
46                           Abram S. Jenks School            252       2520   
47                       Eliza B. Kirkbride School            258       2580   
48                       George W. Nebinger School            259       2590   
49                              Academy at Palumbo            262       2620   
50                                Southwark School            264       2640   
51                      Science Leadership Academy            265       2650   
52                        Constitution High School            267       2670   
53                          Vare-Washington 

* As a note, two schools were added into this score card spreadsheet (from the same data site but separated onto this spreadsheet) and three schools from the original Enrollment and Demographics were not featured on the list.  This all slightly edits the original list of school sites ( might not include this note ).

In [29]:
sms_penn_schools = pd.read_csv('../data/sms_penn_schools.csv')

In [30]:
sms_penn_schools.shape

(65, 337)

In [31]:
sms_penn_schools[['School', 'Grades Served', 'Retention Score', 'Grad Score']]

School Grades Served  \
0                         John Bartram High School        '9-12'   
1                    West Philadelphia High School        '9-12'   
2      Paul Robeson High School for Human Services        '9-12'   
3                     William L. Sayre High School        '9-12'   
4                         William T. Tilden School         '5-8'   
5                           Motivation High School        '9-12'   
6                                John Barry School         'K-8'   
7                         William C. Bryant School         'K-8'   
8                       Joseph W. Catharine School         'K-5'   
9                       Benjamin B. Comegys School         'K-8'   
10                          Sadie Alexander School         'K-8'   
11                          Andrew Hamilton School         'K-8'   
12                      Avery D. Harrington School         'K-8'   
13                             Henry C. Lea School         'K-8'   
14                    William C. Longstreth School         'K-8'   
15                         Morton McMichael School         'K-8'   
16                         S. Weir Mitchell School         'K-8'   
17                         Thomas G. Morton School         'K-5'   
18                             Samuel Powel School         'K-4'   
19                             James Rhoads School         'K-8'   
20                        Martha Washington School         'K-8'   
21                          Add B. Anderson School         'K-8'   
22                              Alain Locke School         'K-8'   
23                      Rudolph Blankenburg School         'K-8'   
24                 Middle Years Alternative School         '5-8'   
25                  South Philadelphia High School        '9-12'   
26                   Benjamin Franklin High School        '9-12'   
27    High School for Creative and Performing Arts        '9-12'   
28        Science Leadership Academy Middle School         '5-6'   
29                  Julia R. Masterman School H.S.        '9-12'   
..                                             ...           ...   
35                 General George A. McCall School         'K-8'   
36                      Delaplaine McDaniel School         'K-8'   
37                      William M. Meredith School         'K-8'   
38                            Robert Morris School         'K-8'   
39              Girard Academic Music Program H.S.        '9-12'   
40              Girard Academic Music Program M.S.         '5-8'   
41                         Edwin M. Stanton School         'K-8'   
42                     Albert M. Greenfield School         'K-8'   
43                        Chester A. Arthur School         'K-8'   
44                          Laura W. Waring School         'K-8'   
45                           Andrew Jackson School         'K-8'   
46                           Abram S. Jenks School         'K-5'   
47                       Eliza B. Kirkbride School         'K-8'   
48                       George W. Nebinger School         'K-8'   
49                              Academy at Palumbo        '9-12'   
50                                Southwark School         'K-8'   
51                      Science Leadership Academy        '9-12'   
52                        Constitution High School        '9-12'   
53                          Vare-Washington School         'K-8'   
54     High School of Engineering and Science H.S.        '9-12'   
55     High School of Engineering and Science M.S.         '7-8'   
56                  Strawberry Mansion High School        '9-12'   
57                          James G. Blaine School         'K-8'   
58                             William Dick School         'K-8'   
59                            Edward Heston School         'K-8'   
60                        William D. Kelley School         'K-8'   
61                  General George G. Meade School         'K-8'   
62  Parkway Center City Middle C

* I once again singled out the metrics; "School" and "Grades Served" along with, "Retention Score" and "Graduation Score".  But this time only looking at 65 schools as opposed to over 300.

* As a note, to make things simpler, I will further refer to these 65 schools as "Penn" schools.

* Below are the top 10 schools of the "Penn" schools with the lowest retention rates.

In [32]:
sms_penn_schools.iloc[[56,3,26,36,59,6,12,23,22,0],[0,257]]

School  Retention Score
56  Strawberry Mansion High School               54
3     William L. Sayre High School               67
26   Benjamin Franklin High School               67
36      Delaplaine McDaniel School               68
59            Edward Heston School               68
6                John Barry School               69
12      Avery D. Harrington School               69
23      Rudolph Blankenburg School               69
22              Alain Locke School               70
0         John Bartram High School               71

Below are the top 10 "Penn" schools with the highest retention rates

In [33]:
sms_penn_schools.iloc[[55,29,39,54,27,30,52,51,49,40],[0,257]]

School  Retention Score
55   High School of Engineering and Science M.S.              100
29                Julia R. Masterman School H.S.               99
39            Girard Academic Music Program H.S.               98
54   High School of Engineering and Science H.S.               98
27  High School for Creative and Performing Arts               98
30                 Julia R. Masterman School M.S               98
52                      Constitution High School               97
51                    Science Leadership Academy               97
49                            Academy at Palumbo               97
40            Girard Academic Music Program M.S.               96

Below are the top 10 "Penn" schools with the lowest graduation rates.

* As a note, only 18 of the schools featured a graduation rate.  Most of these schools being a high school. I still did however take this information into account.  At this point I am also flagging schools that are appearing more than once on a given list.

In [34]:
sms_penn_schools.iloc[[56,3,25,26,63,1,0,5,33],[0,290]]

School Grad Score
56  Strawberry Mansion High School         40
3     William L. Sayre High School         49
25  South Philadelphia High School         53
26   Benjamin Franklin High School         59
63        Parkway West High School         62
1    West Philadelphia High School         66
0         John Bartram High School         67
5           Motivation High School         83
33        Franklin Learning Center         85

* Below are the top 10 "Penn" schools with the highest graduation rates.

In [35]:
sms_penn_schools.iloc[[39,62,29,52,51,54,49,2,27],[0,290]]

School Grad Score
39              Girard Academic Music Program H.S.        100
62  Parkway Center City Middle College High School        100
29                  Julia R. Masterman School H.S.         99
52                        Constitution High School         99
51                      Science Leadership Academy         98
54     High School of Engineering and Science H.S.         98
49                              Academy at Palumbo         96
2      Paul Robeson High School for Human Services         95
27    High School for Creative and Performing Arts         95

* Below are the top 10 "Penn" schools with the lowest percentage of students attending more than 95% of instructional days.

In [36]:
sms_penn_schools.iloc[[1,3,56,26,19,61,12,23,0],[0,243]]

School  Attendance (95%+) Score
1    West Philadelphia High School                       14
3     William L. Sayre High School                       14
56  Strawberry Mansion High School                       14
26   Benjamin Franklin High School                       17
19             James Rhoads School                       21
61  General George G. Meade School                       21
12      Avery D. Harrington School                       24
23      Rudolph Blankenburg School                       24
0         John Bartram High School                       26

* Below are the top 10 "Penn" schools with the highest percentage of students attending more than 95% of instructional days.

In [37]:
sms_penn_schools.iloc[[30,29,55,35,27,10,42,37,40,46], [0,243]]

School  Attendance (95%+) Score
30                 Julia R. Masterman School M.S                       93
29                Julia R. Masterman School H.S.                       92
55   High School of Engineering and Science M.S.                       87
35               General George A. McCall School                       86
27  High School for Creative and Performing Arts                       82
10                        Sadie Alexander School                       80
42                   Albert M. Greenfield School                       80
37                    William M. Meredith School                       79
40            Girard Academic Music Program M.S.                       79
46                         Abram S. Jenks School                       77

* Below are the top 10 "Penn" schools with the lowest percentage of students attending less than 80% of instructional days.

In [38]:
sms_penn_schools.iloc[[29,30,40,10,28,35,37,42,46,47],[0,255]]

School  Attendance (<80%) Score
29            Julia R. Masterman School H.S.                        0
30             Julia R. Masterman School M.S                        0
40        Girard Academic Music Program M.S.                        0
10                    Sadie Alexander School                        1
28  Science Leadership Academy Middle School                        1
35           General George A. McCall School                        1
37                William M. Meredith School                        1
42               Albert M. Greenfield School                        1
46                     Abram S. Jenks School                        1
47                 Eliza B. Kirkbride School                        1

* Below are the top 10 "Penn" schools with the highest percentage of students attending less than 80% of instructional days.

In [39]:
sms_penn_schools.iloc[[3,56,1,26,0,25,61,23,12,59],[0,255]]

School  Attendance (<80%) Score
3     William L. Sayre High School                       49
56  Strawberry Mansion High School                       49
1    West Philadelphia High School                       40
26   Benjamin Franklin High School                       38
0         John Bartram High School                       35
25  South Philadelphia High School                       27
61  General George G. Meade School                       24
23      Rudolph Blankenburg School                       23
12      Avery D. Harrington School                       21
59            Edward Heston School                       21

In [40]:
pd.read_csv('../data/sms_nonpenn_schools.csv')

School SRC School ID  \
0                             John M. Patterson School           140   
1                                       Penrose School           144   
2                                  Furness High School           216   
3                                D. Newlin Fell School           219   
4                               F. Amedee Bregy School           224   
5                                Francis S. Key School           254   
6                           George W. Sharswood School           263   
7             The Science Leadership Academy at Beeber           268   
8                               John H. Taggart School           269   
9                                Overbrook High School           402   
10    Murrell Dobbins Career and Technical High School           406   
11              Lewis C. Cassidy Academics Plus School           424   
12                               Samuel Gompers School           428   
13                          Robert E. Lamberton School           432   
14                         E. Washington Rhodes School           435   
15                         Kenderton Elementary School           436   
16                         Overbrook Elementary School           437   
17                             Thomas M. Peirce School           438   
18                              Dr. Ethel Allen School           444   
19                            Tanner G. Duckrey School           446   
20                            Richard R. Wright School           447   
21                        Overbrook Educational Center           448   
22                        Thomas A. Edison High School           502   
23                       Philadelphia Military Academy           505   
24   Jules E. Mastbaum Area Vocational Technical Hi...           506   
25                       Parkway Northwest High School           507   
26                       William W. Bodine High School           515   
27                             Penn Treaty High School           516   
28                             Penn Treaty High School           516   
29                              Julia de Burgos School           517   
..                                                 ...           ...   
224             Sankofa Freedom Academy Charter School           W99   
225           Franklin Towne Charter Elementary School           X03   
226                      Tacony Academy Charter School           X04   
227                      Tacony Academy Charter School           X04   
228                   Aspira Charter School at Stetson           X06   
229                  Mastery Charter School at Harrity           X07   
230                     Mastery Charter School at Mann           X08   
231                  Mastery Charter School at Smedley           X09   
232                Universal Charter School at Bluford           X10   
233                 Universal Charter School at Daroff           X11   
234                 Mastery Charter School at Douglass           X12   
235                     Aspira Charter School at Olney           X14   
236                   Mastery Charter School at Clymer           X15   
237                    Mastery Charter School at Gratz           X16   
238                    Mastery Charter School at Gratz           X16   
239              Universal Charter School at Audenried           X17   
240                   Universal Charter School at Vare           X18   
241                  Lindley Academy Charter at Birney           X19   
242                Mastery Charter School at Cleveland           X20   
243              Universal Charter School at Creighton           X21   
244  The Philadelphia Charter School for Arts and S...           X22   
245  Memphis Street Academy Charter School at J.P. ...           X23   
246                 Universal Charter School at Alcorn           X24   
247                Mastery Charter School at Pastorius           X26   
248             

* This file contains the remainder of the schools in the Philadelphia district that were not classified as a "Penn" school, otherwise known as not being within the three mile radius of campus.

In [41]:
sms_nonpenn_schools = pd.read_csv('../data/sms_nonpenn_schools.csv')

In [42]:
sms_nonpenn_schools.shape

(254, 337)

In [43]:
sms_nonpenn_schools[['School','Grades Served','Retention Score', 'Grad Score']]

School Grades Served  \
0                             John M. Patterson School         'K-4'   
1                                       Penrose School         'K-8'   
2                                  Furness High School        '9-12'   
3                                D. Newlin Fell School         'K-8'   
4                               F. Amedee Bregy School         'K-8'   
5                                Francis S. Key School         'K-6'   
6                           George W. Sharswood School         'K-8'   
7             The Science Leadership Academy at Beeber        '9-12'   
8                               John H. Taggart School         'K-8'   
9                                Overbrook High School        '9-12'   
10    Murrell Dobbins Career and Technical High School        '9-12'   
11              Lewis C. Cassidy Academics Plus School         'K-8'   
12                               Samuel Gompers School         'K-8'   
13                          Robert E. Lamberton School         'K-8'   
14                         E. Washington Rhodes School         'K-8'   
15                         Kenderton Elementary School         'K-8'   
16                         Overbrook Elementary School         'K-8'   
17                             Thomas M. Peirce School         'K-6'   
18                              Dr. Ethel Allen School         'K-8'   
19                            Tanner G. Duckrey School         'K-8'   
20                            Richard R. Wright School         'K-5'   
21                        Overbrook Educational Center         'K-8'   
22                        Thomas A. Edison High School        '9-12'   
23                       Philadelphia Military Academy        '9-12'   
24   Jules E. Mastbaum Area Vocational Technical Hi...        '9-12'   
25                       Parkway Northwest High School        '9-12'   
26                       William W. Bodine High School        '9-12'   
27                             Penn Treaty High School        '9-12'   
28                             Penn Treaty High School         '6-8'   
29                              Julia de Burgos School         'K-8'   
..                                                 ...           ...   
224             Sankofa Freedom Academy Charter School         'K-8'   
225           Franklin Towne Charter Elementary School         'K-8'   
226                      Tacony Academy Charter School        '9-12'   
227                      Tacony Academy Charter School         'K-8'   
228                   Aspira Charter School at Stetson         '5-8'   
229                  Mastery Charter School at Harrity         'K-8'   
230                     Mastery Charter School at Mann         'K-6'   
231                  Mastery Charter School at Smedley         'K-6'   
232                Universal Charter School at Bluford         'K-6'   
233                 Universal Charter School at Daroff         'K-8'   
234                 Mastery Charter School at Douglass         'K-8'   
235                     Aspira Charter School at Olney        '9-12'   
236                   Mastery Charter School at Clymer         'K-6'   
237                    Mastery Charter School at Gratz        '9-12'   
238                    Mastery Charter School at Gratz         '7-8'   
239              Universal Charter School at Audenried        '9-12'   
240                   Universal Charter School at Vare         '5-8'   
241                  Lindley Academy Charter at Birney         'K-8'   
242                Mastery Charter School at Cleveland         'K-8'   
243              Universal Charter School at Creighton         'K-8'   
244  The Philadelphia Charter School for Arts and S...         'K-8'   
245  Memphis Street Academy Charter School at J.P. ...         '5-8'   
246                 Universal Charter School at Alcorn         'K-8'   
247                Mastery Charter School at Pastorius         'K-8'   
248             

* Once again, I start my exploration by singling out the "School" and the "Grades Served" along with, "Retention Score", and "Graduation Score".  There are 254 schools in this set so it is a bit more to look over and analyze.

* Below, are the top 10 schools of these "Non Penn" schools that have the lowest retention rate of students.

In [44]:
sms_nonpenn_schools.iloc[[79,9,65,188,17,22,56,75,11,78],[0,257]]

School  Retention Score
79                Theodore Roosevelt School               58
9                     Overbrook High School               64
65           Martin Luther King High School               65
188                  Khepera Charter School               67
17                  Thomas M. Peirce School               69
22             Thomas A. Edison High School               69
56                              Building 21               70
75                 John F. McCloskey School               71
11   Lewis C. Cassidy Academics Plus School               72
78                Samuel Pennypacker School               72

* Below, are the top 10 of the "Non Penn" schools that have the highest retention rate of students.

In [45]:
sms_nonpenn_schools.iloc[[61,172,154,173,208,211,225,88,124,167],[0,257]]

School  Retention Score
61                                 Central High School               99
172  Mathematics, Science, and Technology Community...               99
154             Eugenio Maria De Hostos Charter School               98
173  Mathematics, Science, and Technology Community...               98
208        Folk Arts-Cultural Treasures Charter School               98
211                   Northwood Academy Charter School               98
225           Franklin Towne Charter Elementary School               98
88           Academy for the Middle Years at Northwest               97
124                      Arts Academy at Benjamin Rush               97
167                Philadelphia Academy Charter School               94

* Below are the top 10 of the "Non Penn" schools that have the lowest graduation rate of students.

* Similarly with the last data set, not every school had an entry for their graduation rate- only 62 schools had a score.  But I once again took all of these available scores into account.

In [46]:
sms_nonpenn_schools.iloc[[65,9,90,22,237,60,93,239,53,27],[0,290]]

School Grad Score
65          Martin Luther King High School         48
9                    Overbrook High School         50
90                   Frankford High School         55
22            Thomas A. Edison High School         56
237        Mastery Charter School at Gratz         62
60                  Kensington High School         63
93                 Samuel Fels High School         69
239  Universal Charter School at Audenried         69
53      Kensington Health Sciences Academy         71
27                 Penn Treaty High School         72

* Below are the top 10 of the "Non Penn" schools that have the highest graduation rate of students.

In [47]:
sms_nonpenn_schools.iloc[[89,61,124,64,172,226,7,158,26,63],[0,290]]

School Grad Score
89                                Lankenau High School        100
61                                 Central High School        100
124                      Arts Academy at Benjamin Rush         99
64                  Philadelphia High School for Girls         97
172  Mathematics, Science, and Technology Community...         97
226                      Tacony Academy Charter School         96
7             The Science Leadership Academy at Beeber         96
158               Multicultural Academy Charter School         95
26                       William W. Bodine High School         95
63                          Walter B. Saul High School         94

* Below are the top 10 of the "Non Penn" schools that have the highest percentage of students attending more than 95% of instructional days

In [48]:
sms_nonpenn_schools.iloc[[190,219,193,23,188,226,218,227,153,236],[0,243]]

School  \
190  Philadelphia Electrical and Technology Charter...   
219                     Antonia Pantoja Charter School   
193                  Mastery Charter School at Lenfest   
23                       Philadelphia Military Academy   
188                             Khepera Charter School   
226                      Tacony Academy Charter School   
218                Pan American Academy Charter School   
227                      Tacony Academy Charter School   
153                Christopher Columbus Charter School   
236                   Mastery Charter School at Clymer   

     Attendance (95%+) Score  
190                       17  
219                       59  
193                       43  
23                        40  
188                       83  
226                       26  
218                       60  
227                       55  
153                       77  
236                       35

* As I was calculating the data for attendance, I found there to be an error in the spreadsheet I was working with.  I then went back and re-formated the file and re-uploaded it.  Because I am unsure how far back the error impacted my data, I decided to re-run all of the charts for this dataset that I already computed. 

In [49]:
pd.read_csv('../data/updated_sms_nonpenn_schools.csv')

School SRC School ID  \
0                             John M. Patterson School           140   
1                                  Furness High School           216   
2                                D. Newlin Fell School           219   
3                               F. Amedee Bregy School           224   
4                                Francis S. Key School           254   
5                           George W. Sharswood School           263   
6             The Science Leadership Academy at Beeber           268   
7                               John H. Taggart School           269   
8                                Overbrook High School           402   
9     Murrell Dobbins Career and Technical High School           406   
10              Lewis C. Cassidy Academics Plus School           424   
11                               Samuel Gompers School           428   
12                          Robert E. Lamberton School           432   
13                         E. Washington Rhodes School           435   
14                         Kenderton Elementary School           436   
15                         Overbrook Elementary School           437   
16                             Thomas M. Peirce School           438   
17                              Dr. Ethel Allen School           444   
18                            Tanner G. Duckrey School           446   
19                        Overbrook Educational Center           448   
20                      General George G. Meade School           457   
21                        Thomas A. Edison High School           502   
22                       Philadelphia Military Academy           505   
23   Jules E. Mastbaum Area Vocational Technical Hi...           506   
24                       Parkway Northwest High School           507   
25                       William W. Bodine High School           515   
26                             Penn Treaty High School           516   
27                             Penn Treaty High School           516   
28                              Julia de Burgos School           517   
29                             Alexander Adaire School           520   
..                                                 ...           ...   
224             Sankofa Freedom Academy Charter School           W99   
225           Franklin Towne Charter Elementary School           X03   
226                      Tacony Academy Charter School           X04   
227                      Tacony Academy Charter School           X04   
228                   Aspira Charter School at Stetson           X06   
229                  Mastery Charter School at Harrity           X07   
230                     Mastery Charter School at Mann           X08   
231                  Mastery Charter School at Smedley           X09   
232                Universal Charter School at Bluford           X10   
233                 Universal Charter School at Daroff           X11   
234                 Mastery Charter School at Douglass           X12   
235                     Aspira Charter School at Olney           X14   
236                   Mastery Charter School at Clymer           X15   
237                    Mastery Charter School at Gratz           X16   
238                    Mastery Charter School at Gratz           X16   
239              Universal Charter School at Audenried           X17   
240                   Universal Charter School at Vare           X18   
241                  Lindley Academy Charter at Birney           X19   
242                Mastery Charter School at Cleveland           X20   
243              Universal Charter School at Creighton           X21   
244  The Philadelphia Charter School for Arts and S...           X22   
245  Memphis Street Academy Charter School at J.P. ...           X23   
246                 Universal Charter School at Alcorn           X24   
247                Mastery Charter School at Pastorius           X26   
248             

* This is the new updated "Non Penn Schools" file.

In [50]:
upd_sms_nonpenn_schools = pd.read_csv('../data/updated_sms_nonpenn_schools.csv')

In [51]:
upd_sms_nonpenn_schools.shape

(254, 337)

In [52]:
upd_sms_nonpenn_schools[['School', 'Grades Served', 'Retention Score', 'Grad Score']]

School Grades Served  \
0                             John M. Patterson School         'K-4'   
1                                  Furness High School        '9-12'   
2                                D. Newlin Fell School         'K-8'   
3                               F. Amedee Bregy School         'K-8'   
4                                Francis S. Key School         'K-6'   
5                           George W. Sharswood School         'K-8'   
6             The Science Leadership Academy at Beeber        '9-12'   
7                               John H. Taggart School         'K-8'   
8                                Overbrook High School        '9-12'   
9     Murrell Dobbins Career and Technical High School        '9-12'   
10              Lewis C. Cassidy Academics Plus School         'K-8'   
11                               Samuel Gompers School         'K-8'   
12                          Robert E. Lamberton School         'K-8'   
13                         E. Washington Rhodes School         'K-8'   
14                         Kenderton Elementary School         'K-8'   
15                         Overbrook Elementary School         'K-8'   
16                             Thomas M. Peirce School         'K-6'   
17                              Dr. Ethel Allen School         'K-8'   
18                            Tanner G. Duckrey School         'K-8'   
19                        Overbrook Educational Center         'K-8'   
20                      General George G. Meade School         'K-8'   
21                        Thomas A. Edison High School        '9-12'   
22                       Philadelphia Military Academy        '9-12'   
23   Jules E. Mastbaum Area Vocational Technical Hi...        '9-12'   
24                       Parkway Northwest High School        '9-12'   
25                       William W. Bodine High School        '9-12'   
26                             Penn Treaty High School        '9-12'   
27                             Penn Treaty High School         '6-8'   
28                              Julia de Burgos School         'K-8'   
29                             Alexander Adaire School         'K-8'   
..                                                 ...           ...   
224             Sankofa Freedom Academy Charter School         'K-8'   
225           Franklin Towne Charter Elementary School         'K-8'   
226                      Tacony Academy Charter School        '9-12'   
227                      Tacony Academy Charter School         'K-8'   
228                   Aspira Charter School at Stetson         '5-8'   
229                  Mastery Charter School at Harrity         'K-8'   
230                     Mastery Charter School at Mann         'K-6'   
231                  Mastery Charter School at Smedley         'K-6'   
232                Universal Charter School at Bluford         'K-6'   
233                 Universal Charter School at Daroff         'K-8'   
234                 Mastery Charter School at Douglass         'K-8'   
235                     Aspira Charter School at Olney        '9-12'   
236                   Mastery Charter School at Clymer         'K-6'   
237                    Mastery Charter School at Gratz        '9-12'   
238                    Mastery Charter School at Gratz         '7-8'   
239              Universal Charter School at Audenried        '9-12'   
240                   Universal Charter School at Vare         '5-8'   
241                  Lindley Academy Charter at Birney         'K-8'   
242                Mastery Charter School at Cleveland         'K-8'   
243              Universal Charter School at Creighton         'K-8'   
244  The Philadelphia Charter School for Arts and S...         'K-8'   
245  Memphis Street Academy Charter School at J.P. ...         '5-8'   
246                 Universal Charter School at Alcorn         'K-8'   
247                Mastery Charter School at Pastorius         'K-8'   
248             

* Below are the top 10 "Non Penn" schools with the lowest retention rates

In [53]:
upd_sms_nonpenn_schools.iloc[[78,8,64,188,16,21,55,74,10,77],[0,257]]

School  Retention Score
78                Theodore Roosevelt School               58
8                     Overbrook High School               64
64           Martin Luther King High School               65
188                  Khepera Charter School               67
16                  Thomas M. Peirce School               69
21             Thomas A. Edison High School               69
55                              Building 21               70
74                 John F. McCloskey School               71
10   Lewis C. Cassidy Academics Plus School               72
77                Samuel Pennypacker School               72

* Below are the top 10 "Non Penn" schools with the highest retention rates

In [54]:
upd_sms_nonpenn_schools.iloc[[60,172,154,173,208,211,225,87,123,168],[0,257]]

School  Retention Score
60                                 Central High School               99
172  Mathematics, Science, and Technology Community...               99
154             Eugenio Maria De Hostos Charter School               98
173  Mathematics, Science, and Technology Community...               98
208        Folk Arts-Cultural Treasures Charter School               98
211                   Northwood Academy Charter School               98
225           Franklin Towne Charter Elementary School               98
87           Academy for the Middle Years at Northwest               97
123                      Arts Academy at Benjamin Rush               97
168                Philadelphia Academy Charter School               97

* Below are the top 10 "Non Penn" schools with the lowest graduation rates.  
Once again, there were only 62 schools that provided a graduation rate.

In [55]:
upd_sms_nonpenn_schools.iloc[[64,8,89,237,59,92,239,52,235],[0,290]]

School Grad Score
64          Martin Luther King High School         48
8                    Overbrook High School         50
89                   Frankford High School         55
237        Mastery Charter School at Gratz         62
59                  Kensington High School         63
92                 Samuel Fels High School         69
239  Universal Charter School at Audenried         69
52      Kensington Health Sciences Academy         71
235         Aspira Charter School at Olney         71

* Below are the top 10 "Non Penn" schools with the highest graduation rate.

In [56]:
upd_sms_nonpenn_schools.iloc[[88,60,123,172,63,226,6,158,25,62],[0,290]]

School Grad Score
88                                Lankenau High School        100
60                                 Central High School        100
123                      Arts Academy at Benjamin Rush         99
172  Mathematics, Science, and Technology Community...         97
63                  Philadelphia High School for Girls         97
226                      Tacony Academy Charter School         96
6             The Science Leadership Academy at Beeber         96
158               Multicultural Academy Charter School         95
25                       William W. Bodine High School         95
62                          Walter B. Saul High School         94

* Below are the top 10 "Non Penn" schools with the lowest percentage of students attending more then 95% of instructional days.

In [57]:
upd_sms_nonpenn_schools.iloc[[235,8,64,239,190,193,21,18,20,89],[0,243]]

School  \
235                     Aspira Charter School at Olney   
8                                Overbrook High School   
64                      Martin Luther King High School   
239              Universal Charter School at Audenried   
190  Philadelphia Electrical and Technology Charter...   
193                  Mastery Charter School at Lenfest   
21                        Thomas A. Edison High School   
18                            Tanner G. Duckrey School   
20                      General George G. Meade School   
89                               Frankford High School   

     Attendance (95%+) Score  
235                        7  
8                         16  
64                        16  
239                       16  
190                       17  
193                       43  
21                        19  
18                        21  
20                        21  
89                        21

* Below are the top 10 "Non Penn" schools with the highest percentage of students attending more than 95% of instructional days.

In [58]:
upd_sms_nonpenn_schools.iloc[[56,198,60,208,188,204,84,146,99,153],[0,243]]

School  Attendance (95%+) Score
56                                  The U School                       95
198                   Green Woods Charter School                       88
60                           Central High School                       85
208  Folk Arts-Cultural Treasures Charter School                       83
188                       Khepera Charter School                       83
204       Philadelphia Montessori Charter School                       82
84                   Hill-Freedman World Academy                       69
146                      Joseph Greenberg School                       80
99                      Benjamin Franklin School                       80
153          Christopher Columbus Charter School                       77

* Below are the top 10 "Non Penn" schools with the lowest percentage of students attending less than 80% of instructional days.

In [59]:
upd_sms_nonpenn_schools.iloc[[29,146,172,173,205,225,6,56,60,87],[0,255]]

School  \
29                             Alexander Adaire School   
146                            Joseph Greenberg School   
172  Mathematics, Science, and Technology Community...   
173  Mathematics, Science, and Technology Community...   
205                            Ad Prima Charter School   
225           Franklin Towne Charter Elementary School   
6             The Science Leadership Academy at Beeber   
56                                        The U School   
60                                 Central High School   
87           Academy for the Middle Years at Northwest   

     Attendance (<80%) Score  
29                         0  
146                        0  
172                        0  
173                        0  
205                        0  
225                        0  
6                          1  
56                         1  
60                         1  
87                         1

* Below are the top 10 "Non Penn" schools with the highest percentage of students attending less than 80% of instructional days.

In [60]:
upd_sms_nonpenn_schools.iloc[[8,235,64,239,59,21,89,52,92,80],[0,255]]

School  Attendance (<80%) Score
8                    Overbrook High School                       48
235         Aspira Charter School at Olney                       44
64          Martin Luther King High School                       41
239  Universal Charter School at Audenried                       37
59                  Kensington High School                       37
21            Thomas A. Edison High School                       36
89                   Frankford High School                       34
52      Kensington Health Sciences Academy                       26
92                 Samuel Fels High School                       24
80                  Edward T. Steel School                       24